app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Any:
+    all_number = []
+    result = {}
+
+    def check_data(*args: int) -> Callable:
+        if args in all_number:
+            print("Getting from cache")
+        else:
+            result[f"{func.__name__}{args}"] = func(*args)
```

Comment:
```suggestion
            result[args] = func(*args)
```
results will be created separately for every decorated function because it's declared inside the decorator. so concatenating function name and args is redundant